token-lending/program/src/processor.rs

```diff
@@ -1741,6 +1785,93 @@ fn process_liquidate_obligation(
+    }
+
+    let account_info_iter = &mut accounts.iter();
+    let source_liquidity_info = next_account_info(account_info_iter)?;
+    let destination_collateral_info = next_account_info(account_info_iter)?;
+    let destination_liquidity_info = next_account_info(account_info_iter)?;
+    let repay_reserve_info = next_account_info(account_info_iter)?;
+    let repay_reserve_liquidity_supply_info = next_account_info(account_info_iter)?;
+    let withdraw_reserve_info = next_account_info(account_info_iter)?;
+    let withdraw_reserve_collateral_mint_info = next_account_info(account_info_iter)?;
+    let withdraw_reserve_collateral_supply_info = next_account_info(account_info_iter)?;
+    let withdraw_reserve_liquidity_supply_info = next_account_info(account_info_iter)?;
+    let withdraw_reserve_liquidity_fee_receiver_info = next_account_info(account_info_iter)?;
+    let obligation_info = next_account_info(account_info_iter)?;
+    let lending_market_info = next_account_info(account_info_iter)?;
+    let lending_market_authority_info = next_account_info(account_info_iter)?;
+    let user_transfer_authority_info = next_account_info(account_info_iter)?;
+    let clock = &Clock::from_account_info(next_account_info(account_info_iter)?)?;
+    let token_program_id = next_account_info(account_info_iter)?;
+
+    let withdraw_collateral_amount = _liquidate_obligation(
+        program_id,
+        liquidity_amount,
+        source_liquidity_info,
+        destination_collateral_info,
+        repay_reserve_info,
+        repay_reserve_liquidity_supply_info,
+        withdraw_reserve_info,
+        withdraw_reserve_collateral_supply_info,
+        obligation_info,
+        lending_market_info,
+        lending_market_authority_info,
+        user_transfer_authority_info,
+        clock,
+        token_program_id,
+    )?;
+
+    _refresh_reserve_interest(program_id, withdraw_reserve_info, clock)?;
+    let withdraw_liquidity_amount = _redeem_reserve_collateral(
+        program_id,
+        withdraw_collateral_amount,
+        destination_collateral_info,
+        destination_liquidity_info,
+        withdraw_reserve_info,
+        withdraw_reserve_collateral_mint_info,
+        withdraw_reserve_liquidity_supply_info,
+        lending_market_info,
+        lending_market_authority_info,
+        user_transfer_authority_info,
+        clock,
+        token_program_id,
+    )?;
+    let mut withdraw_reserve = Reserve::unpack(&withdraw_reserve_info.data.borrow())?;
+    if &withdraw_reserve.config.fee_receiver != withdraw_reserve_liquidity_fee_receiver_info.key {
+        msg!("Withdraw reserve liquidity fee receiver does not match the reserve liquidity fee receiver provided");
+        return Err(LendingError::InvalidAccountInput.into());
+    }
+    let protocol_fee =
+        withdraw_reserve.calculate_protocol_liquidation_fee(withdraw_liquidity_amount)?;
+
+    spl_token_transfer(TokenTransferParams {
```

Comment:
Maybe for accounts that aren't explicitly checked in this function scope (as in just checked by a different function),
maybe use the variables that they are presumably equal to instead to be a little bit safer.
For example, instead of using `destination_liquidity_info`, can we use `reserve.fee_receiver` or whatever